tests/treemodel/schema/schema.js

```diff
@@ -206,6 +250,52 @@ describe( 'checkQuery', () => {
+	} );
+
+	it( 'should support required attributes', () => {
+		schema.registerItem( 'a', '$inline' );
+		schema.requireAttributes( 'a', [ 'name' ] );
+		schema.requireAttributes( 'a', [ 'href' ] );
+		schema.allow( { name: 'a', inside: '$block', attributes: [ 'name', 'href', 'title', 'target' ] } );
+
+		// Even though a is allowed in $block thanks to inheriting from $inline, we require href or name attribute.
+		expect( schema.checkQuery( { name: 'a', inside: '$block' } ) ).to.be.false;
+
+		// Even though a with title is allowed, we have to meet at least on required attributes set.
+		expect( schema.checkQuery( { name: 'a', inside: '$block', attributes: [ 'title' ] } ) ).to.be.false;
+
+		expect( schema.checkQuery( { name: 'a', inside: '$block', attributes: [ 'name' ] } ) ).to.be.true;
+		expect( schema.checkQuery( { name: 'a', inside: '$block', attributes: [ 'href' ] } ) ).to.be.true;
+		expect( schema.checkQuery( { name: 'a', inside: '$block', attributes: [ 'name', 'href' ] } ) ).to.be.true;
+		expect( schema.checkQuery( { name: 'a', inside: '$block', attributes: [ 'name', 'title', 'target' ] } ) ).to.be.true;
+	} );
+
+	it( 'should support multiple attributes', () => {
+		// Let's take example case, where image item has to have a pair of "alt" and "src" attributes.
+		// Then it could have other attribute which is allowed on inline elements, i.e. "bold".
+		schema.registerItem( 'img', '$inline' );
+		schema.requireAttributes( 'img', [ 'alt', 'src' ] );
+		schema.allow( { name: '$inline', inside: '$block', attributes: 'bold' } );
+		schema.allow( { name: 'img', inside: '$block', attributes: [ 'alt', 'src' ] } );
+
+		// Image without any attributes is not allowed.
+		expect( schema.checkQuery( { name: 'img', inside: '$block', attributes: [ 'alt' ] } ) ).to.be.false;
+
+		// Image can't have just alt or src.
+		expect( schema.checkQuery( { name: 'img', inside: '$block', attributes: [ 'alt' ] } ) ).to.be.false;
+		expect( schema.checkQuery( { name: 'img', inside: '$block', attributes: [ 'src' ] } ) ).to.be.false;
+
+		expect( schema.checkQuery( { name: 'img', inside: '$block', attributes: [ 'alt', 'src' ] } ) ).to.be.true;
+
+		// Because of inherting from $inline, image can have bold
+		expect( schema.checkQuery( { name: 'img', inside: '$block', attributes: [ 'alt', 'src', 'bold' ] } ) ).to.be.true;
+		// But it can't have only bold without alt or/and src.
+		expect( schema.checkQuery( { name: 'img', inside: '$block', attributes: [ 'alt', 'bold' ] } ) ).to.be.false;
+		expect( schema.checkQuery( { name: 'img', inside: '$block', attributes: [ 'src', 'bold' ] } ) ).to.be.false;
+		expect( schema.checkQuery( { name: 'img', inside: '$block', attributes: [ 'bold' ] } ) ).to.be.false;
+
+		// Even if image has src and alt, it can't have attributes that weren't allowed
+		expect( schema.checkQuery( { name: 'img', inside: '$block', attributes: [ 'alt', 'src', 'attr' ] } ) ).to.be.false;
 	} );
```

Comment:
There should be a test to check it `requireAttributes` was called on the parent item it does not affect a child item (not allow nor disallow element with required attributes).